scene/main/node.cpp

```diff
@@ -2416,14 +2396,13 @@ NodePath Node::get_path() const {
 	const Node *n = this;
 
 	Vector<StringName> path;
+	path.resize(data.depth);
 
 	while (n) {
-		path.push_back(n->get_name());
+		path.ptrw()[n->data.depth - 1] = n->get_name();
```

Comment:
`ptrw()` is quite costly and should be cached:

```suggestion
StringName *ptrw = path.ptrw();

while (n) {
		ptrw[n->data.depth - 1] = n->get_name();
```

Reply:
Adjusted. Didn't know that, Thanks!